File: base/animal/.cache/cheat/ts/student.ts
class Animal {
    especie: string;
    barulho: string;
    estagio: number;

    constructor(especie: string, barulho: string) {
        this.especie = especie;
        this.barulho = barulho;
        this.estagio = 0;
    }

    fazerBarulho(): string {
        if (this.estagio === 0) {
            return "---";
        }
        if (this.estagio === 4) {
            return "RIP";
        }
        return this.barulho;
    }

    envelhecer(nivel: number): void {
        this.estagio += nivel;
        if (this.estagio >= 4) {
            console.log(`warning: ${this.especie} morreu`);
            this.estagio = 4;
        }
    }

    toString(): string {
        return `${this.especie}:${this.estagio}:${this.barulho}`;
    }
}

class Adapter {
    animal: Animal;
    constructor(especie: string, barulho: string) {
        this.animal = new Animal(especie, barulho);
    }

    grow(qtd: number): void {
        this.animal.envelhecer(qtd);
    }

    noise(): void {
        console.log(this.animal.fazerBarulho());
    }

    show(): void {
        console.log(this.animal.toString());
    }
}

export {Adapter};

File: base/animal/.cache/cheat/java/Student.java
class Animal {
    private String especie;
    private String barulho;
    private int estagio;

    public Animal(String especie, String barulho) {
        this.especie = especie;
        this.barulho = barulho;
        this.estagio = 0;
    }

    public String fazerBarulho() {
        if (estagio == 0) {
            return "---";
        }
        if (estagio == 4) {
            return "RIP";
        }
        return this.barulho;
    }

    public void envelhecer(int nivel) {
        estagio += nivel;
        if (estagio >= 4) {
            System.out.printf("warning: %s morreu%n", especie);
            estagio = 4;
        }
    }

    public String toString() {
        return String.format("%s:%d:%s", especie, estagio, barulho);
    }
}

public class Student {
    private Animal animal;
    public Student(String especie, String barulho) {
        animal = new Animal(especie, barulho);
    }
    
    public void noise() {
        System.out.println(animal.fazerBarulho());
    }

    public void grow(int nivel) {
        animal.envelhecer(nivel);
    }

    public void show() {
        System.out.println(animal.toString());
    }
}
